### ISSUE TITLE
Cannot add custom certificates for git hosts because user namespace doesn't have expected annotation

### ISSUE THREAD

**Issue by SkorikSergey** [eclipse/che#21327]
### Describe the bug

Tried to add che-git-self-signed-cert  using steps from this [PR-1222](https://github.com/eclipse-che/che-operator/pull/1222). Failed on step 3 because user namespace doesn't have `che.eclipse.org/workspace-namespace-owner-uid` annotation.

### Che version

7.45

### Steps to reproduce

1. Create a config map called che-git-self-signed-cert in the namespace where CheCluster resource is stored with contents like this:
```kind: ConfigMap
apiVersion: v1
metadata:
  name: che-git-self-signed-cert
data:
  githost: somewhere.else
  ca.crt: <<contents of some certificate or any string really for this demonstration purposes >>
```

2. Set the `spec.server.gitSelfSignedCert` property of the CheCluster CR to true.

3. Observe that in all namespaces marked with the `che.eclipse.org/workspace-namespace-owner-uid` annotation, there is a new configmap created with a name based on the name and namespace of the CheCluster resource in the form.

### Expected behavior

In all namespaces marked with the `che.eclipse.org/workspace-namespace-owner-uid` annotation, there is a new configmap created with a name based on the name and namespace of the CheCluster resource in the form

### Runtime

OpenShift

### Installation method

OperatorHub

### Environment

Linux

### Logs
[devworkspace-webhook-server-6dc696f87-g9l2p-webhook-server.log](https://github.com/eclipse/che/files/8462998/devworkspace-webhook-server-6dc696f87-g9l2p-webhook-server.log)
[devworkspace-controller-manager-6c7b746c78-djkvk-devworkspace-controller.log](https://github.com/eclipse/che/files/8462999/devworkspace-controller-manager-6c7b746c78-djkvk-devworkspace-controller.log)

**Comment by ibuziuk** [comment]
@metlos could you please take a look? looks like a missing label during namespace provisioning

**Comment by skabashnyuk** [comment]
@SkorikSergey can you provide steps to reproduce on a new cluster?

**Comment by SkorikSergey** [comment]
1. Start a new cluster(checked on OCP 4.9 and 4.10).
2. Deploy Eclipse Che from OperatorHub(now there is 7.45.0)
3. Add git self signed cert according to this PR description - https://github.com/eclipse-che/che-operator/pull/1222.

**Comment by skabashnyuk** [comment]
During work on https://github.com/eclipse-che/che-operator/pull/1027 the label `che.eclipse.org/workspace-namespace-owner-uid` was set manualy. No automation is available at this point.

**Comment by ibuziuk** [comment]
is the label expected to be added by the admin manually to every single user namespace? Will it be possible to add the label during the namespace provisioning and backport it to 7.44.x / 3.0 ?

**Comment by metlos** [comment]
Currently, this is a manual step https://issues.redhat.com/browse/RHDEVDOCS-3364. I have a draft PR https://github.com/eclipse-che/che-operator/pull/1359 ready that reacts to the namespace labels by default created by the che-server. Currently I have trouble manually testing it

**Comment by skabashnyuk** [comment]
@SkorikSergey https://github.com/eclipse-che/che-operator/pull/1359 has been merged

**Comment by tolusha** [comment]
@SkorikSergey 
The ConfigMap must have `app.kubernetes.io/part-of=che.eclipse.org` label. Otherwise `che-operator` won't recognize it.
Pls see doc https://github.com/eclipse/che/issues/21327#issuecomment-1109816272

**Comment by dmytro-ndp** [comment]
@metlos: thank you for the fix!
[Your commit above](https://github.com/eclipse-che/che-operator/commit/33a59db04798a89b08e0d519f0c718b7ec09e795) should be cherry-picked into 7.16.x branch to be available in DS 3.0.0.

**Comment by metlos** [comment]
> @metlos: thank you for the fix! [Your commit above](https://github.com/eclipse-che/che-operator/commit/33a59db04798a89b08e0d519f0c718b7ec09e795) should be cherry-picked into 7.16.x branch to be available in DS 3.0.0.

@dmytro-ndp is that really `7.16.x` branch? I got asked for `7.46.x` in the chat...

Would you please be so kind and specify all the branches that this should be merged into. I no longer know the release policies in Che that intimately.

**Comment by dmytro-ndp** [comment]
@metlos: sorry for confusing :-|
I meant 7.46.x branch - it's needed to have fix in DS 3.0.0.
